service/src/integrationTest/kotlin/fi/espoo/evaka/occupancy/OccupancyTest.kt

```diff
@@ -0,0 +1,542 @@
+                )
+            )
+
+            it.insertTestCareArea(DevCareArea(id = careArea1, name = "1", shortName = "1"))
+            it.insertTestCareArea(DevCareArea(id = careArea2, name = "2", shortName = "2"))
+
+            it.insertTestDaycare(
+                DevDaycare(
+                    id = daycareInArea1,
+                    areaId = careArea1,
+                    type = setOf(CareType.CENTRE, CareType.PRESCHOOL, CareType.PREPARATORY_EDUCATION)
+                )
+            )
+            it.insertTestDaycareGroup(DevDaycareGroup(id = daycareGroup1, daycareId = daycareInArea1))
+            it.insertTestDaycareGroup(DevDaycareGroup(id = daycareGroup2, daycareId = daycareInArea1))
+            it.insertTestCaretakers(groupId = daycareGroup1, amount = 3.0)
+            it.insertTestCaretakers(groupId = daycareGroup2, amount = 3.0)
+
+            it.insertTestDaycare(
+                DevDaycare(
+                    id = familyUnitInArea2,
+                    areaId = careArea2,
+                    type = setOf(CareType.FAMILY)
+                )
+            )
+            it.insertTestDaycareGroup(DevDaycareGroup(id = familyGroup1, daycareId = familyUnitInArea2))
+            it.insertTestDaycareGroup(DevDaycareGroup(id = familyGroup2, daycareId = familyUnitInArea2))
+            it.insertTestCaretakers(groupId = familyGroup1, amount = 3.0)
+            it.insertTestCaretakers(groupId = familyGroup2, amount = 3.0)
+        }
+    }
+
+    @Test
+    fun `calculateDailyUnitOccupancyValues smoke test`() {
+        db.transaction { tx ->
+            ChildBuilder(tx, today).childOfAge(3)
+                .hasPlacement().ofType(PlacementType.DAYCARE).toUnit(daycareInArea1).fromDay(-1).toDay(0).exec()
+                .withGroupPlacement().toGroup(daycareGroup1).execAndReturn()
+        }
+
+        db.read {
+            val period = FiniteDateRange(today.minusDays(1), today)
+            val occupancyValues =
+                it.calculateDailyUnitOccupancyValues(today, period, OccupancyType.CONFIRMED, areaId = careArea1)
+
+            assertEquals(1, occupancyValues.size)
+            assertEquals(daycareInArea1, occupancyValues[0].key.unitId)
+            assertEquals(2, occupancyValues[0].occupancies.size)
+            assertEquals(1.75, occupancyValues[0].occupancies[period.start]!!.sum)
+            assertEquals(1.0, occupancyValues[0].occupancies[period.end]!!.sum)
+            assertEquals(1, occupancyValues[0].occupancies[period.start]!!.headcount)
+            assertEquals(1, occupancyValues[0].occupancies[period.end]!!.headcount)
+            assertEquals(6.0, occupancyValues[0].occupancies[period.start]!!.caretakers)
+            assertEquals(6.0, occupancyValues[0].occupancies[period.end]!!.caretakers)
+            assertEquals(4.2, occupancyValues[0].occupancies[period.start]!!.percentage)
+            assertEquals(2.4, occupancyValues[0].occupancies[period.end]!!.percentage)
```

Comment:
With all these assertions I feel it would be simpler to just do `assertEquals(listOf(...), occupancyValues)` 😄